momento-sdk/src/main/java/momento/sdk/ScsDataClient.java

```diff
@@ -1890,6 +1923,32 @@ public void onFailure(@Nonnull Throwable e) {
     return returnFuture;
   }
 
+  private CompletableFuture<CacheSortedSetGetRankResponse> sendSortedSetGetRank(
```

Comment:
I figured we can migrate to something like this to cut down on the repetition in the internal client. I could do it all on one line but it is a bit confusing without named arguments.
If there's more I can do to clean this up let me know. I'm sure there's a more object oriented way to encapsulate this that will cut down our 3,000 line client even more.

Reply:
the main thing I have been thinking of for breaking up these clients in the future is pulling the code for each data type out into a separate file; either using abstract classes or decorator pattern or something.  I think that will help a fair amount when we decide it's worth it.